core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.html

```diff
@@ -1,81 +1,216 @@
+        </textarea>
+        <span class="oppia-input-box-subtitle">
+          <i>
+            Topic description should be at most <[MAX_CHARS_IN_TOPIC_DESCRIPTION]> characters.
+          </i>
+        </span>
+        <div>
+          <button class="btn btn-secondary" ng-click="togglePreview()">Show/Hide Preview</button>
+          <div ng-if="topic.getId() && topicPreviewCardIsShown">
+            <preview-thumbnail name="editableName"
+                               description="editableDescription"
+                               preview-footer="topic.getCanonicalStoryIds().length"
+                               filename = "topic.getThumbnailFilename()"
+                               thumbnail-bg-color="topic.getThumbnailBgColor()"
+                               bg-color="'#2F6687'">
+            </preview-thumbnail>
+          </div>
+        </div>
+        <span ng-if="editableDescriptionIsEmpty && topicDescriptionChanged" class="form-text" style="font-size: smaller">
         What does this topic contain?
-      </span>
-    </div>
-
-    <div class="canonical-stories">
-      <label class="form-heading"> Canonical Stories </label>
-      <button class="btn btn-secondary protractor-test-create-story-button"
-              ng-click="createCanonicalStory()"
-              style="float: right;">
-        Create Story
+        </span>
+      </div>
+    </md-card>
+    <md-card class="oppia-subtopics-list-card">
+      <h3>Subtopics</h3>
+      <button class="btn add-subtopic-btn protractor-test-add-subtopic-button" ng-click="createSubtopic()">
+        + ADD SUBTOPIC
       </button>
-      <stories-list ng-if="topic.getCanonicalStoryIds().length > 0"
-                    story-summaries="canonicalStorySummaries"
-                    topic="topic">
-      </stories-list>
-      <h4 ng-if="topic.getCanonicalStoryIds().length === 0" style="color: gray;">
-        Click on Create Story to create a new canonical story for the topic
+      <h4 ng-if="subtopics.length === 0" class="add-entity-text">
+        Click on Add Subtopic to create a new subtopic for the topic.
       </h4>
-    </div>
-  </md-card>
+      <div ng-if="subtopics.length">
+        <div class="list-header">
+          <span>Name</span>
+          <span># of Skills</span>
+        </div>
+        <div ng-repeat="subtopic in subtopics">
+          <div class="subtopics-list-item">
+            <span class="protractor-test-subtopic" ng-click="navigateToSubtopic(subtopic.getId())"><[subtopic.getTitle()]></span>
+            <span><[subtopic.getSkillSummaries().length]>
+              <button class="btn btn-secondary protractor-test-delete-subtopic-button" ng-click="deleteSubtopic(subtopic.getId())">Delete</button>
+            </span>
+          </div>
+        </div>
+      </div>
+    </md-card>
+    <md-card class="oppia-subtopics-list-card">
+      <h3>Skills</h3>
+      <button class="btn add-subtopic-btn" ng-click="createSkill()">
+        + ADD SKILL
```

Comment:
Add Skill doesn't update the skills list in the topic editor, as the skill opens in a new tab. So, this leads to version error if I try editing in the topic editor after adding a skill, without refreshing the page. Can you fix that? (Fetch the topic again from the backend, after a skill is added so the local version is the correct one)

Reply:
Yeah i also encountered this(Sorry I though refreshing the page was the ideal scenario). Anyways, fixed now!

Reply:
Not completely fixed. Although the skill is added correctly, if there are pending changes (not saved) in the topic editor before creating a skill (like new subtopic or edited topic description), when a skill is created, they suddenly disappear and reappear on saving the changes at this point. Can you prevent the initial disappearance?
You could also maybe force the user to save all changes before creating a skill as well from the topic editor (we do that before creating a story).

Reply:
I did the same thing we follow for story creation(force to save changes before creating skill).